### PR TITLE
feat: Resync resources in the namespaces on NamespaceCredential changes

### PR DESCRIPTION
**Pull request by maciaszczykm**

### CODE REVIEW

File: controller/internal/controller/cluster_controller.go [hunk 1]
```diff
@@ -42,7 +43,24 @@ type ClusterReconciler struct {
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		WithOptions(controller.Options{MaxConcurrentReconciles: 1}). // Hard requirement for current namespace credentials implementation.
+		// Setting max concurrent reconciles is a hard requirement for current namespace credentials implementation.
+		// Following watch ensures that if namespaced credentials change, all objects that use them will be reconciled.
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
+		Watches(&v1alpha1.NamespaceCredentials{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, nc client.Object) []reconcile.Request {
```
Comment (michaeljguarino) [inline controller/internal/controller/cluster_controller.go:49]: Is there no way to DRY up these EnqueueRequestsFromMapFunc handlers?  Maybe generics could help or something?

Comment (maciaszczykm) [reply controller/internal/controller/cluster_controller.go:49]: I will give it another try as at the first attempt I had problem with interface for list object that client uses to list resources.

**Comment by maciaszczykm** [comment]
Converting to draft to connect secret with namespace credentials, so we will have full event propagation for token changes.

File: controller/internal/controller/cluster_controller.go (hunk 1 above)
Comment (maciaszczykm) [reply controller/internal/controller/cluster_controller.go]: It's done now.

**Comment by maciaszczykm** [comment]
Added logic to recreate service account secret if it gets deleted.

**Pull request merged**